### PR TITLE
Fix Pilot port number config for Ingress and ConfigMap

### PR DESCRIPTION
**Pull request by myidpt**
Partially fixes https://github.com/istio/istio/issues/3757.
Pilot starts without cert ready in auth tests caused Envoy unable to connect to Pilot. This fixes the issue for prow e2e tests. Also need to fix the same issue on other tests not using istio-auth.yaml.

### CODE REVIEW

**Comment by mandarjog** [comment]
```
2018-03-08T01:35:12.369997Z	info	Envoy command: [-c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 2 --parent-shutdown-time-s 3 --service-cluster istio-ingress --service-node ingress~~istio-ingress-66fb78d8c4-tdmj5.simple-auth-test-2a76e7f23f334f1f922f~simple-auth-test-2a76e7f23f334f1f922f.svc.cluster.local --max-obj-name-len 189 -l info --v2-config-only]
[2018-03-08 01:35:12.396][18][info][main] external/envoy/source/server/server.cc:184] initializing epoch 0 (hot restart version=9.200.16384.256.options=capacity=16384, num_slots=8209 hash=228984379728933363)
[2018-03-08 01:35:12.404][18][critical][main] external/envoy/source/server/server.cc:71] error initializing configuration '/etc/istio/proxy/envoy-rev0.json': unable to read file: /etc/certs/root-cert.pem
2018-03-08T01:35:12.405985Z	warn	Epoch 0 terminated with an error: exit status 1
```
@myidpt  @wattli  do not merge ?
eventually 
```
018-03-08T01:36:24.249865Z	info	Envoy command: [-c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 2 --parent-shutdown-time-s 3 --service-cluster istio-ingress --service-node ingress~~istio-ingress-66fb78d8c4-tdmj5.simple-auth-test-2a76e7f23f334f1f922f~simple-auth-test-2a76e7f23f334f1f922f.svc.cluster.local --max-obj-name-len 189 -l info --v2-config-only]
[2018-03-08 01:36:24.261][50][info][main] external/envoy/source/server/server.cc:184] initializing epoch 0 (hot restart version=9.200.16384.256.options=capacity=16384, num_slots=8209 hash=228984379728933363)
[2018-03-08 01:36:24.265][50][info][config] external/envoy/source/server/configuration_impl.cc:52] loading 0 listener(s)
[2018-03-08 01:36:24.265][50][info][config] external/envoy/source/server/configuration_impl.cc:92] loading tracing configuration
[2018-03-08 01:36:24.265][50][info][config] external/envoy/source/server/configuration_impl.cc:106]   loading tracing driver: envoy.zipkin
[2018-03-08 01:36:24.265][50][info][config] external/envoy/source/server/configuration_impl.cc:119] loading stats sink configuration
[2018-03-08 01:36:24.265][50][info][main] external/envoy/source/server/server.cc:359] starting main dispatch loop
[2018-03-08 01:36:24.288][50][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:128] cm init: initializing cds
[2018-03-08 01:36:24.291][50][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:132] cm init: all clusters initialized
[2018-03-08 01:36:24.291][50][info][main] external/envoy/source/server/server.cc:343] all clusters initialized. initializing init manager
[2018-03-08 01:36:24.292][50][info][config] external/envoy/source/server/listener_manager_impl.cc:579] all dependencies initialized. starting workers
[2018-03-08 01:36:24.292][50][info][upstream] external/envoy/source/server/lds_subscription.cc:70] lds: fetch failure: network error
[2018-03-08 01:36:25.771][50][info][upstream] external/envoy/source/server/lds_subscription.cc:70] lds: fetch failure: network error
[2018-03-08 01:36:26.813][50][info][upstream] external/envoy/source/server/lds_subscription.cc:70] lds: fetch failure: network error
```

**Comment by myidpt** [comment]
@mandarjog How did you get the Envoy log?

**Comment by myidpt** [comment]
@ldemailly I found it on prow, thanks.

**Comment by myidpt** [comment]
I saw the TestSimpleIngress still exhaust the retries. I will try to flip "optional: true" for secret mount on ingress (and maybe also others) to see the result.

**Comment by myidpt** [comment]
I can exclude the Envoy restarting as an issue. This test fails without Pilot's Envoy restarting:
https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4066/e2e-simpleTests/834/
Pilot's Envoy log
https://storage.googleapis.com/istio-prow/pull/istio_istio/4066/e2e-simpleTests/834/artifacts/simple-auth-test-e544e2a624f94902b4f9/istio-pilot-6d87788d8c-ckp8q_container:istio-proxy.log

Still investigating.

**Comment by kyessenov** [comment]
You could dump bootstrap and certs, and start it locally, right?

File: pilot/pkg/proxy/envoy/v1/watcher.go [hunk 1]
```diff
@@ -85,6 +85,8 @@ func (w *watcher) Run(ctx context.Context) {
 	// agent consumes notifications from the controller
 	go w.agent.Run(ctx)
 
+	time.Sleep(time.Second * 30)
```
Comment (ldemailly) [inline pilot/pkg/proxy/envoy/v1/watcher.go]: this needs to be a wait for the cert not a sleep

Comment (myidpt) [reply pilot/pkg/proxy/envoy/v1/watcher.go]: that's just experimental :) I fixed it.

**Comment by myidpt** [comment]
Thanks to Kuat. We found the issue is due to ingress discovery-address is set to 8080:
https://storage.googleapis.com/istio-prow/pull/istio_istio/4122/e2e-simpleTests/837/artifacts/simple-auth-test-4fc511534e9c42c7955e/yaml/istio-one-namespace-auth.yaml

**Comment by ldemailly** [comment]
the cert and startup order still is an issue, glad you find both

**Comment by mandarjog** [comment]
That is great sleuth work @kuat and @myipdt. 
More reasons to move off updateVersion to helm!

**Comment by kyessenov** [comment]
Having one way is always better, so helm or not helm, get rid of the
multitude of ways we create YAMLs :)

**Comment by ldemailly** [comment]
While the port number is clearly the most egregious bug and it's fantastic to solve it asap

I'd like to not lose the fact that #3757 is how that bug sneaked in undetected behind the "oh ingress fails already anyway" - so we do need to start pilot's envoy only when the certs are there and get those certs there faster by starting CA + service accounts first.

**Comment by myidpt** [comment]
@ldemailly I will have a PR to fix the proxy bootstrapping issue as well.

**Comment by mandarjog** [comment]
@ldemailly https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_istio/4066/e2e-suite-rbac-auth/1019/
Fails TestVMExpansion....

**Pull request merged**